File: dodam-system-domain/dodam-domain-rds/src/main/java/b1nd/dodam/domain/rds/notice/repository/NoticeFileRepository.java
package b1nd.dodam.domain.rds.notice.repository;

import b1nd.dodam.domain.rds.notice.entity.Notice;
import b1nd.dodam.domain.rds.notice.entity.NoticeFile;
import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface NoticeFileRepository extends JpaRepository<NoticeFile, Long> {

    @EntityGraph(attributePaths = {"notice"})
    List<NoticeFile> findAllByNoticeIn(List<Notice> notices);

}

File: services/service-notice/src/main/kotlin/com/b1nd/dodamdodam/notice/domain/notice/repository/NoticeFileRepository.kt
package com.b1nd.dodamdodam.notice.domain.notice.repository

import com.b1nd.dodamdodam.notice.domain.notice.entity.NoticeEntity
import com.b1nd.dodamdodam.notice.domain.notice.entity.NoticeFileEntity
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository

interface NoticeFileRepository : JpaRepository<NoticeFileEntity, Long> {

    @EntityGraph(attributePaths = ["notice"])
    fun findAllByNoticeIn(notices: List<NoticeEntity>): List<NoticeFileEntity>

    fun deleteAllByNotice(notice: NoticeEntity)
}
